### PR TITLE
gh-124217: Add RFC 9637 reserved IPv6 block `3fff::/20`

### PR DESCRIPTION
**Pull request by y5c4l3**
https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml

* Issue: gh-124217

### CODE REVIEW

**Review by ZeroIntensity — APPROVED** [review]
LGTM, tests are all passing and there's nothing here that I'm concerned about.

File: Lib/test/test_ipaddress.py [hunk 1]
```diff
@@ -2433,6 +2433,8 @@ def testReservedIpv6(self):
         self.assertTrue(ipaddress.ip_address('2001:30::').is_global)
         self.assertFalse(ipaddress.ip_address('2001:40::').is_global)
         self.assertFalse(ipaddress.ip_address('2002::').is_global)
+        # issue #124217: conform with RFC 9637
```
Comment (vstinner) [inline Lib/test/test_ipaddress.py]:
```suggestion
        # gh-124217: conform with RFC 9637
```

File: Lib/ipaddress.py [hunk 2]
```diff
@@ -2377,6 +2377,7 @@ class _IPv6Constants:
         IPv6Network('2002::/16'),
         IPv6Network('fc00::/7'),
         IPv6Network('fe80::/10'),
+        IPv6Network('3fff::/20'),
```
Comment (vstinner) [inline Lib/ipaddress.py]: Can you please add a comment with a reference to RFC 9637?

Can you move it between 2002 and fc00?

File: Misc/NEWS.d/next/Library/2024-09-19-20-15-00.gh-issue-124217.j0KlQB.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Add RFC 9637 reserved IPv6 block ``3fff::/20`` in :mod:`ipaddress` module
```
Comment (vstinner) [inline Misc/NEWS.d/next/Library/2024-09-19-20-15-00.gh-issue-124217.j0KlQB.rst]:
```suggestion
Add RFC 9637 reserved IPv6 block ``3fff::/20`` in :mod:`ipaddress` module.
```

Comment (y5c4l3) [reply Misc/NEWS.d/next/Library/2024-09-19-20-15-00.gh-issue-124217.j0KlQB.rst]: Resolved, thanks.

**Review by vstinner — APPROVED** [review]
LGTM. Thanks for the update.

**Comment by vstinner** [comment]
Merged, thanks.

**Pull request merged**